src/handlers/RPInitiatedLogoutRequest.js

```diff
@@ -40,20 +41,17 @@ class RPInitiatedLogoutRequest extends BaseRequest {
    * @returns {Promise}
    */
   static handle (req, res, provider) {
-    const { host } = provider
     const request = new RPInitiatedLogoutRequest(req, res, provider)
 
     return Promise
       .resolve(request)
       .then(request.validate)
-
       // From: https://openid.net/specs/openid-connect-session-1_0.html#RPLogout
       // At the logout endpoint, the OP SHOULD ask the End-User whether they want
       // to log out of the OP as well. If the End-User says "yes", then the OP
       // MUST log out the End-User.
-      .then(host.logout)
-
-      .then(request.redirectToGoodbye.bind(request))
+      .then(request.clearUserSession.bind(request))
```

Comment:
Ah, ok, so it seems like you're shifting clearing the user session from the `host.logout` callback to `clearUserSession` on this request. However, that's not the right place to do it - this request has no idea of what session or cookie mechanism the host provider is using. That's why it's letting it do its own session clearing in `host.logout(request)`.